PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+        toPositions = (cellCoords, cellCenter)
+        # determine the top left and bottom right positions
+        startPosition = [self.HEAP_X0, 0, self.HEAP_X0, 0]  # Color
+        startPosition = add_vector(startPosition, (0, 0, self.CELL_WIDTH, self.CELL_HEIGHT))  # color`
+        cellPair = self.createCellValue(startPosition, val)
+        callEnviron.add(cellPair)
+        self.moveItemsTo(cellPair, toPositions, steps=self.CELL_HEIGHT, sleepTime=0.01)
+
+        # add a new Drawable with the new value, color, and display objects
+        d = drawable(
+            val, self.canvas.itemconfigure(cellPair[0], 'fill')[-1], *cellPair)
+        self.list.append(d)  # store item at the end of the list
+
+        if len(self.list) <= 0:  # heap condition.  The root node, i = 0,
+            return
+
+        for i in range(len(self.list) - 1, 0, -1):
+            if self.list[i] > self.list[(i - 1) // 2]:  # if i is less than its parent
+                self.swap(i, (i - 1) // 2)
+
+        callEnviron.discard(cellPair)
+        # finish the animation
+        self.cleanUp(callEnviron)
+
+    # makes a new heap of size 2
+    def newArray(self):
+        # gets rid of old elements in the list
+        del self.list[:]
+        self.size = 2
+        self.display()
+        # make a new arrow pointing to the top of the Heap
+        self.indexDisplay = self.createIndex(len(self.list))
+
+    def peek(self):
+        self.startAnimations()
+        callEnviron = self.createCallEnvironment()
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        spacing = self.CELL_WIDTH * 3 // 4
+        padding = 10
+        boxSize = 1
+        peekIndex = 0
+        # changed the values that were added to the variables to get the output box in a desirable location
+        outputBox = self.canvas.create_rectangle(
+            (self.HEAP_X0 * 2 + padding),
+            canvasDimensions[1] - self.CELL_WIDTH - padding,
+            (self.HEAP_X0 * 2 + boxSize * spacing + padding * 2),
+            canvasDimensions[1] - padding,
+            fill=self.OPERATIONS_BG)
+        callEnviron.add(outputBox)
+
+        outputBoxCoords = self.canvas.coords(outputBox)
+        midOutputBox = (outputBoxCoords[3] + outputBoxCoords[1]) // 2
+        # create the value to move to output box
+        valueOutput = self.copyCanvasItem(self.list[peekIndex].display_val)
+        valueList = (valueOutput,)
+        callEnviron.add(valueOutput)
+        # move value to output box
+        # self.highlightCodeTags('print', callEnviron)
+        toPositions = (outputBoxCoords[0] + padding / 2 + (peekIndex + 1 / 2) * spacing,
+                        midOutputBox)
+        self.moveItemsTo(valueList, (toPositions,), sleepTime=.02)
+        # make the value 25% smaller
+        newSize = (self.VALUE_FONT[0], int(self.VALUE_FONT[1] * .75))
+        self.canvas.itemconfig(valueOutput, font=newSize)

```

Comment:
A better name for this variable would be `newFont`.  And it is not 25% smaller in this case.  That was true in Array.py module, but not for the heap.